src/ert/ensemble_evaluator/monitor.py

```diff
@@ -101,24 +101,39 @@ async def signal_done(self) -> None:
         )
         logger.debug(f"monitor-{self._id} informed server monitor is done")
 
-    async def track(self) -> AsyncGenerator[CloudEvent, None]:
-        timeout: Optional[float] = None
+    async def track(
+        self, heartbeat_interval: Optional[float] = None
+    ) -> AsyncGenerator[Optional[CloudEvent], None]:
+        """Yield events from the internal event queue with optional heartbeats.
+
+        Heartbeats are represented by None being yielded.
+
+        Heartbeats stops being emitted after a CloseTrackerEvent is found."""
+        _heartbeat_interval: Optional[float] = heartbeat_interval
+        closetracker_received: bool = False
         while True:
             try:
-                # We need a timeout in case TERMINATED event get not sent
-                # when signalling cancel or done to evaluator
-                event = await asyncio.wait_for(self._event_queue.get(), timeout)
+                event = await asyncio.wait_for(
+                    self._event_queue.get(), timeout=_heartbeat_interval
+                )
             except asyncio.TimeoutError:
-                logger.error("Evaluator did not send the TERMINATED event!")
-                break
+                if closetracker_received:
+                    logger.error("Evaluator did not send the TERMINATED event!")
+                    break
+                event = None
             if isinstance(event, CloseTrackerEvent):
-                timeout = self._receiver_timeout
+                closetracker_received = True
+                _heartbeat_interval = self._receiver_timeout
```

Comment:
Just wondering if we need the `receiver_timeout` at all when having the heartbeat in the first place.

Reply:
the heartbeat_interval is `None` by default, so there is not really a heartbeat in place unless you ask for it. 

The receiver timeout is 60 second, while a heartbeat is very short. I thought that this could risk logging the error "Evalutor did not send the TERMINATED event" too often and when it should not.

Reply:
There is only one consumer of the `track()` function except for tests, so the heartbeat could be there by default. It would require modifications in ca 15 tests though to ignore the None event, but that is doable.

Reply:
I see. Then it makes sense to keep it - maybe. Nevertheless the `None` there is only for the sake of tests. It very much depends how trivial this modification of tests would be.

Reply:
Let's keep it for now. I'll create an issue if it is doable to remove it.